docs/de/docs/python-types.md

```diff
@@ -0,0 +1,314 @@
+
+Dies ist auch der häufigste Bereich, in dem sie mit **FastAPI** verwendet werden.
+
+### Einfache Typen
+
+Sie können alle Standard Python-Typen deklarieren, nicht nur `str`.
+
+Sie können z.B. folgende verwenden:
+
+* `int`
+* `float`
+* `bool`
+* `bytes`
+
+```Python hl_lines="1"
+{!../../../docs_src/python_types/tutorial005.py!}
+```
+
+### Generische Typen mit Typ-Parametern
+
+Es gibt einige Datenstrukturen, die andere Werte enthalten können, wie `dict`, `list`, `set` und `tuple`. Und die internen Werte können auch ihren eigenen Typ haben.
+
+Um diese Typen und die internen Typen zu deklarieren, können Sie das Standard Python-Modul `typing` verwenden.
+
+Es existiert speziell für die Unterstützung dieser Typ-Hinweise.
+
+#### `List`
+
+Definieren wir zum Beispiel eine Variable, die eine `list` von `str` sein soll.
+
+Von `typing`, importieren wir `List` (mit dem Großbuchstaben `L`):
+
+```Python hl_lines="1"
+{!../../../docs_src/python_types/tutorial006.py!}
+```
+
+Deklarieren Sie die Variable, mit der gleichen Doppelpunkt (`:`) Syntax.
+
+Als Typ geben Sie `List` an.
+
+Da die Liste ein Typ ist, der einige interne Typen enthält, setzen Sie diese in eckige Klammern:
+
+```Python hl_lines="4"
+{!../../../docs_src/python_types/tutorial006.py!}
+```
+
+!!! tip
+    Die internen Typen in den eckigen Klammern werden als "Typparameter" bezeichnet.
+
+    In diesem Fall ist `str` der Typparameter, der an `List` übergeben wird.
+
+Das bedeutet: "die Variable `items` ist eine `list`, und jedes der Elemente in dieser Liste ist ein `str`".
+
+Auf diese Weise kann Ihr Editor auch bei der Bearbeitung von Einträgen aus der Liste unterstützen:
+
+<img src="/img/python-types/image05.png">
+
+Ohne Typen ist das fast unmöglich zu erreichen.
+
+Beachten Sie, dass die Variable `item` eines der Elemente in der Liste `items` ist.
+
+Und trotzdem weiß der Editor, dass es sich um eine `str` handelt, und bietet entsprechende Unterstützung.
+
+#### `Tuple` und `Set`
+
+Das Gleiche gilt für die Deklaration von `Tuple` und `Set`:
+
+```Python hl_lines="1  4"
+{!../../../docs_src/python_types/tutorial007.py!}
+```
+
+Das bedeutet:
+
+* Die Variable `items_t` ist ein `Tupel` mit 3 Elementen, einem `int`, einem weiteren `int` und einem `str`.
+* Die Variable `items_s` ist ein `set`, und jedes ihrer Elemente ist vom Typ `bytes`.
+
+#### `Dict`
+
+Um ein `dict` zu definieren, übergeben Sie 2 Typ-Parameter, getrennt durch Kommas.
+
+Der erste Typ-Parameter ist für die Schlüssel des `dict`.
+
+Der zweite Typ-Parameter ist für die Werte des `dict`:
+
+```Python hl_lines="1  4"
+{!../../../docs_src/python_types/tutorial008.py!}
+```
+
+Dies bedeutet:
+    
+* Die Variable `prices` ist ein `dict`:
+    * Die Schlüssel dieses `dict` sind vom Typ `str` (z.B. die Namen der einzelnen Artikel).
+    * Die Werte dieses `dict` sind vom Typ `float` (z.B. der Preis jedes Artikels).
+
```

Comment:
The section about `Union`is missing.
Suggestion:

#### Union

You can declare that a variable can be any of **several types**, for example, an `int` or a `str`.

In Python 3.6 and above (including Python 3.10) you can use the `Union` type from `typing` and put inside the square brackets the possible types to accept.

In Python 3.10 there's also an **alternative syntax** were you can put the possible types separated by a <abbr title='also called "bitwise or operator", but that meaning is not relevant here'>vertical bar (`|`)</abbr>.

=== "Python 3.6 and above"

    ```Python hl_lines="1  4"
    {!> ../../../docs_src/python_types/tutorial008b.py!}
    ```

=== "Python 3.10 and above"

    ```Python hl_lines="1"
    {!> ../../../docs_src/python_types/tutorial008b_py310.py!}
    ```

In beiden Fällen bedeutet dies, dass `item` ein `int` oder ein `str` sein kann.